ql/src/java/org/apache/hadoop/hive/ql/plan/LoadTableDesc.java

```diff
@@ -157,6 +159,19 @@ public LoadTableDesc(final Path sourcePath,
     }
   }
 
+  public LoadTableDesc(Path path, Table tableHandle, boolean isOverWrite, boolean isIcebergLoad) {
+    super(path, AcidUtils.Operation.NOT_ACID);
+    this.mdTable = tableHandle;
+    this.isIcebergLoad = isIcebergLoad;
+    this.loadFileType = isOverWrite ? LoadFileType.REPLACE_ALL : LoadFileType.KEEP_EXISTING;
+    this.table = Utilities.getTableDesc(tableHandle);
+  }
+
+  @Explain(displayName = "Iceberg Table Load")
+  public boolean isIcebergLoad() {
```

Comment:
And i also think the naming is too iceberg specific.  Although maybe only iceberg handler can load  data by this way right now, we can not make sure other storage handler can also do this.  Therefore, could you rename it to express more generic?